appIcons.js

```diff
@@ -1741,59 +1741,82 @@ var MyShowAppsIconMenu = Utils.defineClass({
     _dtpRedisplay: function() {
         this.removeAll();
 
-        if (this.sourceActor != Main.layoutManager.dummyCursor) { 
-            let powerSettingsMenuItem = this._appendMenuItem(_('Power options'));
-            powerSettingsMenuItem.connect('activate', function () {
-                Util.spawn(['gnome-control-center', 'power']);
+        // true if `which` can find the app
+        function _checkExists(app) {
+            let cmd = "which '" + app + "'";
+            let out = GLib.spawn_command_line_sync(cmd);
+
+            // out contains 1: stdout, 2: stderr, 3: exit code
+            return out[3] == 0;
+        }
+
+        // Only add menu entries for commands that exist in path
+        function _appendItem(obj, info) {
+            if (_checkExists(info.cmd[0])) {
+                let item = obj._appendMenuItem(_(info.title));
```

Comment:
I was surprised to see that all texts were correctly extracted for translation here! Good to know!